tests/utils/common/common.py

```diff
@@ -32,6 +28,79 @@
 import traceback
 
 
+class Result:
+    def __init__(self, stdout, stderr, exited):
+        self.stdout = stdout
+        self.stderr = stderr
+        self.exited = exited
+        self.return_code = exited
+
+
+class Connection:
+    def __init__(self, host, user, port, connect_timeout, connect_kwargs={}):
+        self.host = host
+        self.user = user
+        self.port = port
+        self.connect_timeout = connect_timeout
+        self.connect_kwargs = connect_kwargs
+
+        self.key_filename = None
+
+        for k in connect_kwargs.keys():
```

Comment:
This looks a bit odd...

What if `k==key_filename` is not the first key?

Why not do: 
```
if "key_filename" not in connect_kwargs.keys():
...
else:
```

Reply:
> What if `k==key_filename` is not the first key?

If it exists at all, it must be the first key, because no other key is allowed.

I could use your approach, but it would require three ifs, not two. There are three cases: `key_filename` is there, `key_filename` is not there, or some other key is there (which is an error). Think of it more like arguments parsing, maybe then it looks more logical to use a loop.

Reply:
why not index the key array at `[0]`? I feel like I'm missing something :thinking:

Reply:
I can, but to me it feels unnatural. If we add a flag later, you can just add an additional `if`. With the `[0]` approach this won't work, and then it would probably be reworked into a loop anyway.